### DISCUSSION TITLE
Complejidad LEAVE

### DISCUSSION THREAD

**Discussion by ilpeter** [IIC2133-PUC/2024-2#125] [Tarea 1] [answered]
Leí en otro issue que las complejidades no son solo para lo pedido en el evento, por ejemplo encontrar y eliminar al pinguino en LEAVE, si no también para todo lo otro asociado a los eventos como reordenar los heaps. Al eliminar el pinguino más triste este puede estar en cualquier nodo de el ultimo nivel de el max heap y en cualquier nodo de el ultimo nivel de un heap completo decreciente, encontrarlo para arreglar el hecho de que se vaya no podría ser menor a O(n), por lo que tengo entendido. La complejidad es correcta entonces para LEAVE O(log(n) + log(s))??

**Comment by Segismundios** [comment, marked as answer]
Hola @ilpeter. La complejidad pedida es correcta, se puede hacer el LEAVE y todo lo asociado en O(log(n)+log(s)).
